webserver/webserver/src/main/java/io/helidon/webserver/RequestRouting.java

```diff
@@ -82,12 +82,16 @@ public void route(BareRequest bareRequest, BareResponse bareResponse) {
     }
 
     private static String canonicalize(String p) {
-        String result = p;
-        if (p.charAt(p.length() - 1) == '/') {
-            result = p.substring(0, p.length() - 1);
-        }
-        if (result.isEmpty()) {
+        String result;
```

Comment:
Two things:
* Should this not have a unit test to avoid issues going forward?
* Any reason why not a more fail-fast approach which is easier to read:
```
    private static String canonicalize(String p) {
        if (p == null || p.isEmpty() || p.equals("/")) {
            return "/";
        }
        int lastCharIndex = p.length() - 1;
        if (p.charAt(lastCharIndex) == '/') {
            return p.substring(0, lastCharIndex);
        }
        return p;
    }
```
I'm interested to know what the benefit is of indenting more or delaying the `return` of the method when you already know the result.